tests/scripts/thread-cert/test_history_tracker.py

```diff
@@ -0,0 +1,206 @@
+
+        child_rloc16 = child.get_addr16()
+        leader_rloc16 = leader.get_addr16()
+
+        # Verify the `netinfo` history on child
+
+        netinfo = child.history_netinfo(2)
+        self.assertEqual(len(netinfo), 2)
+        self.assertEqual(netinfo[0]['role'], 'child')
+        self.assertEqual(netinfo[0]['mode'], 'n')
+        self.assertEqual(int(netinfo[0]['rloc16'], 16), child_rloc16)
+        self.assertEqual(netinfo[0]['partition-id'], leader.get_partition_id())
+        self.assertEqual(netinfo[1]['role'], 'detached')
+
+        # Change the child mode and verify that `netinfo` history
+        # records this change.
+
+        child.set_mode('rn')
+        self.simulator.go(SHORT_WAIT)
+        netinfo = child.history_netinfo(1)
+        self.assertEqual(len(netinfo), 1)
+        self.assertEqual(netinfo[0]['mode'], 'rn')
+
+        # Ping from leader to child and check the RX and TX history
+        # on child and leader.
+
+        child_mleid = child.get_mleid()
+        leader_mleid = leader.get_mleid()
+
+        ping_sizes = [10, 100, 1000]
+        num_msgs = len(ping_sizes)
+
+        for size in ping_sizes:
+            leader.ping(child_mleid, size=size)
+
+        leader_tx = leader.history_tx(num_msgs)
+        leader_rx = leader.history_rx(num_msgs)
+        child_tx = child.history_tx(num_msgs)
+        child_rx = child.history_rx(num_msgs)
+
+        for index in range(num_msgs):
+            self.assertEqual(leader_tx[index]['type'], 'ICMP6(EchoReqst)')
+            self.assertEqual(leader_tx[index]['sec'], 'yes')
+            self.assertEqual(leader_tx[index]['prio'], 'norm')
+            self.assertEqual(leader_tx[index]['tx-success'], 'yes')
+            self.assertEqual(leader_tx[index]['radio'], '15.4')
+            self.assertEqual(int(leader_tx[index]['to'], 16), child_rloc16)
+            self.assertEqual(leader_tx[index]['src'][1:-3], leader_mleid)
+            self.assertEqual(leader_tx[index]['dst'][1:-3], child_mleid)
+
+            self.assertEqual(child_rx[index]['type'], 'ICMP6(EchoReqst)')
+            self.assertEqual(child_rx[index]['sec'], 'yes')
+            self.assertEqual(child_rx[index]['prio'], 'norm')
+            self.assertEqual(child_rx[index]['radio'], '15.4')
+            self.assertEqual(int(child_rx[index]['from'], 16), leader_rloc16)
+            self.assertEqual(child_rx[index]['src'][1:-3], leader_mleid)
+            self.assertEqual(child_rx[index]['dst'][1:-3], child_mleid)
+
+            self.assertEqual(leader_rx[index]['type'], 'ICMP6(EchoReply)')
+            self.assertEqual(child_tx[index]['type'], 'ICMP6(EchoReply)')
+
+            self.assertEqual(leader_tx[index]['len'], child_rx[index]['len'])
+            self.assertEqual(leader_rx[index]['len'], child_tx[index]['len'])
+
+
+if __name__ == '__main__':
+    # Skip the test under distcheck build. Error code
+    # 77 means skip this test case in automake tests
+    if os.getenv('DISTCHECK_BUILD') == '1':
```

Comment:
It may be better to have a `TODO` or `FIXME` here to explain why the test case cannot work with distcheck build.

Reply:
Extended the commit in the new push.